### PR TITLE
[10] PATCH /todos/:id

### PR DESCRIPTION
**Pull request by pdany1116**
#### Description

Update existing todos via PATCH/PUT request on /todos/:id.

## Steps to Test or Reproduce

1. Follow test steps in README.
2. All tests should pass.

### CODE REVIEW

File: app/controllers/todos_controller.rb [hunk 1]
```diff
@@ -23,11 +23,26 @@ def create
     render json: '', status: :unprocessable_entity
   end
 
+  def update
+    todo = Todo.find(params[:id])
+    todo.update!(update_todo_params)
```
Comment (cazacutudor) [inline app/controllers/todos_controller.rb:42]: We should be able to chain those commands.

File: features/step_definitions/todo_steps.rb [hunk 2]
```diff
@@ -20,6 +20,9 @@
   get_todo(@last_todo_id)
 end
 
+When 'I update that todo' do
+  update_todo(@last_todo_id)
```
Comment (cazacutudor) [inline features/step_definitions/todo_steps.rb]: We should be more specific with the field that's being updated ?

Comment (cazacutudor) [reply features/step_definitions/todo_steps.rb]: https://cucumber.io/docs/cucumber/cucumber-expressions/

File: features/todo.feature [hunk 3]
```diff
@@ -15,3 +15,9 @@ Feature: Todo API
   Scenario: Create a todo
     When I create a todo
     Then I will get a response with the created todo
+
+  Scenario: Update a todo
+    When I create a todo
```
Comment (cazacutudor) [inline features/todo.feature]: https://cucumber.io/docs/gherkin/reference/#given

File: spec/controllers/todos_controller_spec.rb [hunk 4]
```diff
@@ -126,4 +126,100 @@
       end
     end
   end
+
+  describe '#update' do
+    subject { patch :update, params: }
+
+    let(:params) { default_params }
+    let(:default_params) do
```
Comment (cazacutudor) [inline spec/controllers/todos_controller_spec.rb]: Why do we need this alias ?

File: config/initializers/cors.rb [hunk 5]
```diff
@@ -13,6 +13,6 @@
 
     resource '*',
              headers: :any,
-             methods: %i[get post put delete]
+             methods: %i[get post put delete patch]
```
Comment (cazacutudor) [inline config/initializers/cors.rb:16]: :+1:

**Review by cazacutudor — CHANGES_REQUESTED** [review]

File: spec/controllers/todos_controller_spec.rb (hunk 4 above)
Comment (pdany1116) [reply spec/controllers/todos_controller_spec.rb]: 805fad88f45edfc2709c9382bf82bd125265c227

File: features/todo.feature (hunk 3 above)
Comment (pdany1116) [reply features/todo.feature]: f8b524767b8696e30e12820161eaf896c2bd37a6

File: features/step_definitions/todo_steps.rb (hunk 2 above)
Comment (pdany1116) [reply features/step_definitions/todo_steps.rb]: f8b524767b8696e30e12820161eaf896c2bd37a6

File: app/controllers/todos_controller.rb (hunk 1 above)
Comment (pdany1116) [reply app/controllers/todos_controller.rb:42]: I could do this in one line, but I think it is uglier. `todo = Todo.find(params[:id]).tap { _1.update!(update_todo_params) }`.

**Review by cazacutudor — APPROVED** [review]

**Pull request merged**